get_http.go

```diff
@@ -117,13 +118,54 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 		g.Client = httpClient
 	}
 
-	resp, err := g.Client.Get(u.String())
+	// check to see whether user has specified a range of bytes to download.
+	// if user has, but the range is invalid, fall back to downloading the
+	// whole file
+	byteRange, rangeErr := GetByteRange(u)
+	if rangeErr != nil {
+		if strings.Compare("No byte range provided", rangeErr.Error()) != 0 {
+			fmt.Printf("%s; going to download entire file without a range request",
+				rangeErr)
+		}
+	}
+
+	req, err := http.NewRequest("HEAD", u.String(), nil)
```

Comment:
Took me a second to realize so I'd comment here that you're allocating this up front because if it is a range request, we need to set `Range` headers on a shared request object.